src/main/java/org/logstash/beats/ConnectionHandler.java

```diff
@@ -0,0 +1,112 @@
+package org.logstash.beats;
+
+import io.netty.channel.ChannelDuplexHandler;
+import io.netty.channel.ChannelFuture;
+import io.netty.channel.ChannelFutureListener;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.handler.timeout.IdleState;
+import io.netty.handler.timeout.IdleStateEvent;
+import io.netty.util.AttributeKey;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * Manages the connection state to the beats client.
+ */
+public class ConnectionHandler extends ChannelDuplexHandler {
```

Comment:
I actually like the new name 'ConnectionHandler' better, since the class is really a 'KeepAliveHandler' and an 'IdleConnectionHandler'.  

Unless you have a strong opinion otherwise, I would like to keep as 'ConnectionHandler'

Reply:
I guess thats ok, given the behavior when the `ALL_IDLE` event triggers, closing when everyone is idle (although I'm not quite sure if that will trigger if we are sending keep alive acks every 5 seconds...

Reply:
> although I'm not quite sure if that will trigger if we are sending keep alive acks every 5 seconds...

It won't. The ALL_IDLE will only fire when there is NOTHING read or written.. keep alives are counted as writes.